### PR TITLE
Added support for mocking classes/interfaces with reference returning methods (Closes #139)

### LINKED ISSUE #139: Fatal error when mocking reference returning interface methods
Example:

    interface A
    {
        public function &foo ($x, $y, $z);
    }
    
    $prophet = new Prophecy\Prophet;
    
    $x = $prophet->prophesize();
    $x->willImplement ('A');
    $z = $x->reveal();

This will throw a PHP fatal error:

    PHP Fatal error:  Declaration of Double\A\P1::blah() must be compatible with & A::blah($x, $y, $z) in /home/edson/tmp/vendor/phpspec/prophecy/src/Prophecy/Doubler/Generator/ClassCreator.php(49) : eval()'d code on line 2
    PHP Stack trace:
    PHP   1. {main}() /home/edson/tmp/test.php:0
    PHP   2. Prophecy\Prophecy\ObjectProphecy->reveal() /home/edson/tmp/test.php:14
    PHP   3. Prophecy\Doubler\LazyDouble->getInstance() /home/edson/tmp/vendor/phpspec/prophecy/src/Prophecy/Prophecy/ObjectProphecy.php:105
    PHP   4. Prophecy\Doubler\Doubler->double() /home/edson/tmp/vendor/phpspec/prophecy/src/Prophecy/Doubler/LazyDouble.php:122
    PHP   5. Prophecy\Doubler\Doubler->createDoubleClass() /home/edson/tmp/vendor/phpspec/prophecy/src/Prophecy/Doubler/Doubler.php:105
    PHP   6. Prophecy\Doubler\Generator\ClassCreator->create() /home/edson/tmp/vendor/phpspec/prophecy/src/Prophecy/Doubler/Doubler.php:142
    PHP   7. eval() /home/edson/tmp/vendor/phpspec/prophecy/src/Prophecy/Doubler/Generator/ClassCreator.php:49
    
    shell returned 255

### PR DESCRIPTION
**Pull request by edsonmedina**
Closes #139

### CODE REVIEW

File: src/Prophecy/Doubler/Generator/Node/MethodNode.php [hunk 1]
```diff
@@ -99,6 +110,11 @@ public function setCode($code)
 
     public function getCode()
     {
+        if ($this->returnsReference)
+        {
+            return "throw new \Prophecy\Exception\Doubler\MethodNotFoundException('Returning by reference not supported', get_class(\$this), '{$this->name}');";
```
Comment (everzet) [inline src/Prophecy/Doubler/Generator/Node/MethodNode.php]: `MethodNotFoundException` is the wrong exception for what's happening. You need a custom `ReturnByReferenceException` or something like that.

**Comment by edsonmedina** [comment]
Done

File: src/Prophecy/Doubler/Generator/Node/MethodNode.php (hunk 1 above)
Comment (edsonmedina) [inline src/Prophecy/Doubler/Generator/Node/MethodNode.php]: This has been fixed.

**Comment by everzet** [comment]
Thank you very much for contribution!

**Pull request merged**